### DISCUSSION TITLE
Settings loading as null unless I access them first.

### DISCUSSION THREAD

**Discussion by argonzalez2016** [spatie/laravel-settings#75] [Q&A]
When I load a settings class, my settings are null until I access at least one of the properties first. How do I fix this?

Example:
`$setting = app(SettingClass::class);
dump(empty($setting->someSetting))`
This outputs True.

However when i do it this way
`$setting = app(SettingClass::class)
echo $setting->someSetting;
dump(empty($setting->someSetting));`
I get the proper value: False.

Any help would be appreciated!

**Comment by argonzalez2016** [comment]
Actually this only happens when I use empty(). When I use is_null() everything works as intended.

**Reply by rubenvanassche** [reply]
Hi @argonzalez2016,

Good catch, we don't load settings until they are needed, so in this case `empty` was correctly true because settings weren't loaded. This is fixed in version 2.1.9.
